src/redux/offersSlice.js

```diff
@@ -0,0 +1,52 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getOffers } from 'api';
+
+const initialState = {
+  offers: [],
+  pauseOffers: [],
+  pauseOfferIDs: [],
+  loading: false,
+  error: null
+};
+
+export const fetchOffers = createAsyncThunk(
+  'offers/fetchOffers',
+  async thunkAPI => {
+    try {
+      const result = await getOffers();
+      return result;
+    } catch (err) {
+      return thunkAPI.rejectWithValue(err.message);
+    }
+  }
+);
+
+export const offersSlice = createSlice({
+  name: 'offers',
+  initialState,
+  extraReducers: {
+    [fetchOffers.pending]: state => {
+      state.loading = true;
+    },
+    [fetchOffers.fulfilled]: (state, { payload }) => {
+      state.loading = false;
+      state.offers = payload;
+      state.pauseOffers = Array.isArray(payload)
+        ? payload.filter(
+            offer => offer.externalProperties.PAUSE_OFFER === 'true'
+          )
+        : [];
```

Comment:
is there an option that payload is not an array?

Reply:
payload.filter({externalProperties} => externalProperties.PAUSE_OFFER === 'true')